### ISSUE TITLE
ESLint 2.x support

### ISSUE THREAD

**Issue by SkaterDad** [cyclejs/eslint-config-cycle#3]
[Official ESLint Migration Guide](http://eslint.org/docs/user-guide/migrating-to-2.0.0)

Looks like ESLint 2.x has some significant changes so this config no longer works.

**Comment by staltz** [comment]
We're in the process of migrating most of our repos to TypeScript, so ESLint may not be a must soon. We'll keep on using older version of ESLint in the time being, because honestly it still works and fulfills our needs.

**Comment by SteveALee** [comment]
I guess this config is really for project contributions? The repos don't say. However I discovered using https://github.com/kahlil/cycle-browserify-boilerplate and thought it a good idea to use in my project.
Obviously each project will have own preferences (I was using Hapi) but some may like to extend this.

I guess a workaround is just copy to my project's config. Is there a description anyway of key style guidelines?

I understand ESLint are working on TS support, including plugins, but for now that means using TSLint.

**Comment by SteveALee** [comment]
So I copied across to my client eslint config which uses eslint 2.xx

I had to make these changes:

    "keyword-spacing": [2, { "before": true, "after": true, "overrides": {}}],
    "no-shadow": [2, { "builtinGlobals": false, "hoist": "functions", "allow": [] }],

and deleted the duplicate

"no-cond-assign": 2,

I had to comment out the brace style cutom rule as not sure how to use it locally

**Comment by kahlil** [comment]
@staltz this is good to know. I guess in that case I will also go back to eslint 1.x.x in the the browserify boilerplate and then move to TypeScript as well.

Question though: does using TypeScript also allow for code style linting? I feel strongly about having a community-wide standard for code style.

**Comment by staltz** [comment]
> I guess this config is really for project contributions? The repos don't say. 

That's really the idea, sorry if it's not clear. Nothing stops people from using those code styles, really, even if they are only meant for Cycle libs, but I don't intend to maintain them, because of what I said in this issue: https://github.com/kahlil/cycle-browserify-boilerplate/issues/3

> @kahlil I feel strongly about having a community-wide standard for code style.

Agreed, and we can look into that separately, but I don't feel that eslint-plugin-cycle is recommendable for the general audience. For instance, it doesn't exactly account for the common style of Cycle.js code we see in the wild.

**Comment by SteveALee** [comment]
> For instance, it doesn't exactly account for the common style of Cycle.js code we see in the wild.

Including the videos on egghead.io :)

I agree. I'm afraid I started this off by deciding it would be good to use the same style for my project, but then you immediately hit the semicolons.

Personally it seems a shame you're intending to use traditional class based OOP but I understand TS pushes you that way. I certainly don't want that for my project. Prototypes and object.assign work for me.

**Comment by staltz** [comment]
> Personally it seems a shame you're intending to use traditional class based OOP

It has nothing to do with preference. Building it with classes (or prototype-based classes) is better for performance because it reduces the number of closures accessed. TS itself doesn't push us to use classes.

**Comment by kahlil** [comment]
Ah OK so that means I can kick out the plugin all together anyway right? Just use the eslint config...

**Comment by SteveALee** [comment]
That's what I did.

**Comment by kahlil** [comment]
OK I'll remove it :)

**Comment by kahlil** [comment]
Oh but eslint-config-cycle depends on eslint-plugin-cycle. So how do you guys feel about using https://github.com/feross/standard as a coding style for apps?

**Comment by SteveALee** [comment]
It's easy to use but also insists on no semicolons which is contentious. Otherwise it has the advantage of not having so many rules so leaving more open cycle users. There is a fork that uses semicolons.

Personally I've chosen [air bnb's core](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb) configure for my client and will see how it goes.

**Comment by kahlil** [comment]
Yes I am actually also a big fan of air bnb's core configuration. 

I am actually proposing standard because eslint-cycle-config also does not use semicolons and because standard it is a community effort to bring one code style to JavaScript which I really like.

**Comment by SteveALee** [comment]
Moving my comments over to kahlil/cycle-browserify-boilerplate#3 as @staltz has indicated this repo will be depreciated

**Comment by kahlil** [comment]
Good idea. :)
